akka-remote/src/main/scala/akka/remote/artery/compress/TopHeavyHitters.scala

```diff
@@ -38,10 +38,9 @@ private[remote] final class TopHeavyHitters[T >: Null](val max: Int)(implicit cl
   /** Returns copy(!) of items which are currently considered to be heavy hitters. */
   def snapshot: Array[T] = {
     val snap = Array.ofDim(max).asInstanceOf[Array[T]]
-    var i: Int = 0
-    val heapIterator = heap.reverseIterator
-    while (heapIterator.hasNext) {
-      val index = heapIterator.next()
+    var i = 0
+    while (i < max) {
+      val index = heap(i)
```

Comment:
Initialy heap contains `-1` and this is min heap. The lowest value are on lower indexes so, taking heap in reverse order would generate array with higher value first. 

If for compression is used fixed length number encoding this is irrelevant. But if encoding would have variable length encoding better to have frequently appeared values on lower.